src/p2p/net_node.inl

```diff
@@ -1381,8 +1382,8 @@ namespace nodetool
 
     if(just_take_peerlist)
     {
-      zone.m_net_server.get_config_object().close(con->m_connection_id);
       LOG_DEBUG_CC(*con, "CONNECTION HANDSHAKED OK AND CLOSED.");
+      zone.m_net_server.get_config_object().close(con->m_connection_id);
```

Comment:
this change seems gratuitous, and also makes the log message a lie,
since the message is logged before the actual close(). Likewise for the
change below at line 1448.

Reply:
The log will dereference the context to prints its address, and the context would be technically deleted. But you're right that it's better to copy the address first, keep the order, and print the copied address directoy in the log. I'll do that.